apps/stage-web/src/components/Layouts/MobileInteractiveArea.vue

```diff
@@ -40,6 +40,16 @@ const { send, onAfterMessageComposed, discoverToolsCompatibility } = useChatStor
 const { messages } = storeToRefs(useChatStore())
 const { t } = useI18n()
 
+function isMobileDevice() {
+  return /Mobi|Android|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent)
+}
```

Comment:
```suggestion
function isMobileDevice() {
  return "virtualKeyboard" in navigator
}
```

https://developer.mozilla.org/en-US/docs/Web/HTTP/Guides/Browser_detection_using_the_user_agent
https://developer.mozilla.org/en-US/docs/Web/API/VirtualKeyboard_API

Let's use feature detection instead of UA parse, shall we?

Reply:
Sure.